wagtailtinymce/rich_text.py

```diff
@@ -32,7 +32,92 @@
 from wagtail.utils.widgets import WidgetWithScript
 from wagtail.wagtailadmin.edit_handlers import RichTextFieldPanel
 from wagtail.wagtailcore.rich_text import DbWhitelister
-from wagtail.wagtailcore.rich_text import expand_db_html
+from wagtail.wagtailcore.rich_text import expand_db_html, get_link_handler, get_embed_handler
+from wagtail.wagtailcore.whitelist import allow_without_attributes, attribute_rule, check_url
+
+
+class DbTinymceWhitelister(DbWhitelister):
```

Comment:
The recommended method for updating the whitelisted is to use a  [construct-whitelister-element-rules hook](http://docs.wagtail.io/en/v1.9/reference/hooks.html?highlight=attribute_rule#construct-whitelister-element-rules) - this would remove the need for the `DbWhitelister ` subclass 

@hooks.register('construct_whitelister_element_rules')
def whitelister_element_rules():
    common = {
        'style': True,
        'width': True,
        'margin-left': True,
        'margin-right': True,
        'height': True,
        'border-color': True,
        'text-align': True,
        'background-color': True,
        'vertical-align': True,
        'font-family': True,
        'valign': True,
    }
    table_rule = attribute_rule(dict(common, **{
        'border': True,
        'cellpadding': True,
        'cellspacing': True,
    }))
    row_rule = attribute_rule(common)
    cell_rule = attribute_rule(dict(common, **{
        'colspan': True,
        'scope': True,
        'rowspan': True,
    }))

    return {
        'table': table_rule,
        'thead': allow_without_attributes,
        'tfoot': allow_without_attributes,
        'tbody': allow_without_attributes,
        'colgroup': allow_without_attributes,
        'col': allow_without_attributes,
        'caption': allow_without_attributes,
        'tr': row_rule,
        'th': cell_rule,
        'td': cell_rule,
    }